geonode/qgis_server/views.py

```diff
@@ -319,16 +322,28 @@ def qgis_server_request(request):
     # We have some shortcuts here instead of asking QGIS-Server.
     if params.get('SERVICE') == 'WMS':
         if params.get('REQUEST') == 'GetLegendGraphic':
-            if not params.get('LAYERS'):
-                raise Http404('LAYERS is not found for a GetLegendGraphic')
-            layer = get_object_or_404(Layer, name=params.get('LAYERS'))
+            layer_name = params.get('LAYER')
+            if not layer_name:
+                raise Http404('LAYER is not found for a GetLegendGraphic')
+            layer = get_object_or_404(Layer, name=layer_name)
             return legend(request, layername=layer.name)
 
     # if not shortcut, we forward any request to internal QGIS Server
     qgis_server_url = qgis_server_endpoint(internal=True)
     response = requests.get(qgis_server_url, params)
+
+    content = response.content
+
+    # if it is GetCapabilities request, we need to replace all reference to
+    # our proxy
+    if params.get('REQUEST') == 'GetCapabilities':
```

Comment:
Why do you filter only on the GetCapabilities?
We need to replace in all requests if there is a reference to our internat qgis url.